setup.py

```diff
@@ -15,35 +13,46 @@
     "xarray",
 ]
 
-test_requirements = ["pytest", "pytest-subtests", "serialbox", "coverage"]
-ndsl_requirements = ["ndsl @ git+https://github.com/NOAA-GFDL/NDSL.git@develop"]
-develop_requirements = test_requirements + ndsl_requirements + ["pre-commit"]
+test_requirements = [
+    "coverage",
+    "pytest",
+    "pytest-subtests",
+    "serialbox",
+]
+
+ndsl_requirements = ["ndsl @ git+https://github.com/NOAA-GFDL/NDSL.git@2024.09.00"]
+
+develop_requirements = [
+    *ndsl_requirements,
+    *test_requirements,
+    "pre-commit",
+]
 
 extras_requires = {
-    "test": test_requirements,
-    "ndsl": ndsl_requirements,
     "develop": develop_requirements,
+    "ndsl": ndsl_requirements,
+    "test": test_requirements,
 }
 
 setup(
-    author="The Allen Institute for Artificial Intelligence",
+    author="NOAA - Geophysical Fluid Dynamics Laboratory",
     author_email="oliver.elbert@noaa.gov",
-    python_requires=">=3.11",
+    python_requires=">=3.11,<3.12",
     classifiers=[
         "Development Status :: 2 - Pre-Alpha",
         "Intended Audience :: Developers",
-        "License :: OSI Approved :: BSD License",
+        "License :: OSI Approved :: GPLv3 License",
         "Natural Language :: English",
         "Programming Language :: Python :: 3",
         "Programming Language :: Python :: 3.11",
     ],
-    description="pyFV3 is a NDSL-based FV3 dynamical core for atmospheric models",
+    description="PyFV3 is a NDSL-based FV3 dynamical core for atmospheric models.",
     install_requires=requirements,
     extras_require=extras_requires,
-    license="BSD license",
+    license="GPLv3 license",
     long_description=readme,
     include_package_data=True,
-    name="pyFV3",
+    name="PyFV3",
```

Comment:
Good catch, Rusty. I shouldn't have changed the package name. Will revert the name.

To explain what these two lines do:
- The package name is used when writing imports like `from pyFV3 import ...`. We should thus keep `"pyFV3"` as-is.
- The include list of `find_namespace_packages()` expects folder names/paths to be included. Since the folder is called `"pyFV3"` (with lowercase p), we shouldn't change that either.